test/functional/autocmd/channel_spec.lua

```diff
@@ -0,0 +1,52 @@
+local helpers = require('test.functional.helpers')(after_each)
+local Screen = require('test.functional.ui.screen')
+local clear = helpers.clear
+local command = helpers.command
+local funcs = helpers.funcs
+local meths = helpers.meths
+
+describe('v:event is immutable #18063', function()
+  local screen
+
+  before_each(function()
+    clear()
+    screen = Screen.new(80, 8)
+    screen:set_default_attr_ids({
+      [1] = {bold = true, foreground = Screen.colors.Blue},  -- NonText
+      [2] = {bold = true, reverse = true},  -- MsgSeparator
+      [3] = {background = Screen.colors.Red, foreground = Screen.colors.White},  -- ErrorMsg
+      [4] = {bold = true, foreground = Screen.colors.SeaGreen},  -- MoreMsg
+    })
+    screen:attach()
+  end)
+
+  it('in ChanOpen event', function()
+    command('autocmd ChanOpen * let v:event.info.id = 0')
+    funcs.jobstart({'cat'})
+    screen:expect([[
+                                                                                      |
+      {1:~                                                                               }|
+      {1:~                                                                               }|
+      {1:~                                                                               }|
+      {2:                                                                                }|
+      {3:Error detected while processing ChanOpen Autocommands for "*":}                  |
+      {3:E46: Cannot change read-only variable "v:event.info"}                            |
+      {4:Press ENTER or type command to continue}^                                         |
+    ]])
+  end)
+
+  it('in ChanInfo event', function()
+    command('autocmd ChanInfo * let v:event.info.id = 0')
+    meths.set_client_info('foo', {}, 'remote', {}, {})
+    screen:expect([[
```

Comment:
screen:expect() is a kinda verbose and inefficient way to test this, no?

i'd also consider putting these tests in `test/functional/autocmd/autocmd_spec.lua` since they are really just testing a property of `v:event` for many different cases.

Reply:
I guess I can use a `<CR>` to suppress the hit-enter prompt, then evaluate `v:errmsg`.

Reply:
Oh, I forgot that hit-enter prompt is skipped when there is no UI attached.